src/mount/readdata.cc

```diff
@@ -762,16 +773,31 @@ int read_to_buffer(ReadRecord *rrec, uint64_t current_offset,
 				return SAUNAFS_ERROR_IO;
 			}
 		} catch (Exception &ex) {
+			if (gShowMessagesOnNotificationArea && errorInodeFullPath.empty()) {
+				fs_fullpath(rrec->inode, 0, 0, errorInodeFullPath);
+			}
 			if (try_counter > 0) {
 				print_error_msg(reader, try_counter, ex);
 			}
 			force_prepare = true;
 			if (try_counter > maxRetries) {
+				if (gShowMessagesOnNotificationArea) {
+					ShowMessageOnNotificationArea(
+					    "Read error: Exceeded max retries:" +
+					    std::string(ex.what()) + " " +
+					    errorInodeFullPath.c_str());
```

Comment:
The function c_str() is redundant in this case:
```suggestion
					    errorInodeFullPath);
```

Reply:
Done